validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/SlotBasedScheduledDuties.java

```diff
@@ -43,6 +46,14 @@ public SlotBasedScheduledDuties(
     this.dutyFunction = dutyFunction;
   }
 
+  public Optional<UInt64> getNextAttestationProductionDutyScheduledSlot() {
+    return productionDuties.entrySet().stream()
+        .filter(e -> e.getValue().getType() == DutyType.ATTESTATION_PRODUCTION)
+        .map(Entry::getKey)
+        .sorted()
+        .findFirst();
+  }
```

Comment:
just so we have a brief note, we talked about ways to make this less expensive, we should do that before we merge..